### DISCUSSION TITLE
Style of a table's cell based on the content of another cel?

### DISCUSSION THREAD

**Discussion by jolitp** [typst/typst#4555] [Q&A] [answered]
Suppose I have the following table:

| n   | exp.| result |
|-----|-------| ------|
| 1   | one | one |
| 2   | two | one |

I want the row `| 1   | one | one |` styled as green, and the `| 2   | two | one |` styled as red.

How to go about styling it? Show rule on the table or the cell? Any clever function needed?

**Comment by frozolotl** [comment, marked as answer]
You can't (easily) use a show rule here, but you can instead input the cells pre-colored:
```typ
#let data = (
  ("one", "one"),
  ("one", "two"),
)
#table(
  columns: 3,
  table.header[*n*][*exp.*][*result*],
  ..data
    .enumerate(start: 1)
    .map(((n, (left, right))) => {
      let cell = table.cell.with(fill: if left == right { green } else { red })
      (cell[#n], cell(left), cell(right))
    })
    .flatten(),
)
```

**Reply by jolitp** [reply]
That did the trick. Thank you very much.
